frontend/src/components/projects/ProjectEditContent.tsx

```diff
@@ -214,6 +214,9 @@ function serialize(values) {
   else
     newValues.targeted_end_date = null
 
+  if (!newValues.external_id || newValues.external_id.length == 0)
+    newValues.external_id = null
+
```

Comment:
so external_id either a an empty string, number 0 or an empty array?

Reply:
> so external_id is either a an empty string, a number 0, a null or an empty array?
    newValues.external_id = null
The form uses empty string as a default to an empty string instead of null. The backend accepts an empty string as a valid blank value. The ordering make a mess of things ...